docs/Guide/preconditions/reporting-precondition-failure.mdx

```diff
@@ -0,0 +1,129 @@
+
+There are many possibilities for what you can do with the error, but the simplest is to just send it directly to the
+user. That is what we'll do in this example:
+
+```typescript ts2esm2cjs|{6}|{6}
+import { Listener, UserError, CommandDeniedPayload } from '@sapphire/framework';
+
+export class CommandDeniedListener extends Listener {
+  public run(error: UserError, { message }: CommandDeniedPayload) {
+    return message.channel.send(error.message);
+  }
+}
+```
+
+## Ignoring Precondition Failures
+
+If someone who isn't a bot owner tries to use a command intended only for the bot owner, sometimes you don't want to
+send a message notifying them that they don't have permission. Instead, you'd rather let the command be blocked
+silently. To do this, we can make use of the [`context`][context] property of `UserError`s. This property aims to
+contain information about the context in which the error was thrown, and the value can be absolutely anything.
+
+We can take advantage of this by adding `context: { silent: true }` to the [`this.error()`][preconditionerror] options.
+We'll use the `OwnerOnly` precondition we made in [Creating Preconditions][creating-preconditions] to demonstrate this.
+
+```typescript ts2esm2cjs|{9}|{9}
+import { Precondition, Message } from '@sapphire/framework';
+
+export class OwnerOnlyPrecondition extends Precondition {
+  public run(message: Message) {
+    return message.author.id === 'YOUR_ID'
+      ? this.ok()
+      : this.error({
+          message: 'Only the bot owner can use this command!'
+          context: { silent: true }
+        })
+  }
+}
+```
+
+We can then check if this property exists on the error in our listener, and ignore the failure if we find it.
+
+<Tabs groupId="language-choice">
+<TabItem value="javascript" label="JavaScript" default>
+
+```javascript {5}
+const { Listener } = require('@sapphire/framework');
+
+module.exports = class CommandDeniedListener extends Listener {
+  run(error, { message }) {
+    if (error.context.silent) return;
+    return message.channel.send(error.message);
+  }
+};
+```
+
+</TabItem>
+<TabItem value="ESM" label="ESM">
+
+```javascript {5}
+import { Listener } from '@sapphire/framework';
+
+export class CommandDeniedListener extends Listener {
+  run(error, { message }) {
+    if (error.context.silent) return;
+    return message.channel.send(error.message);
+  }
+}
+```
+
+</TabItem>
+<TabItem value="typescript" label="TypeScript">
+
+```typescript {8}
+import { UserError, CommandDeniedPayload, Listener } from '@sapphire/framework';
+
+export class CommandDeniedListener extends Listener {
+  public run(error: UserError, { message }: CommandDeniedPayload) {
+    // A bit of a hack is required for TypeScript
+    // Since `error.context` is of type `unknown`
+    if (Reflect.get(Object(error.context), 'silent')) return;
+    return message.channel.send(error.message);
+  }
+}
+```
+
+</TabItem>
+</Tabs>
```

Comment:
Tabs left here, replace with `ts2esm2cjs`

Reply:
The problem is that the typescript code isn't the same as the javascript

Reply:
Uh.... hmm.. well Reflect.get works just fine in JS as well so lets do it like so:

---

```ts

import { UserError, CommandDeniedPayload, Listener } from '@sapphire/framework';

export class CommandDeniedListener extends Listener {
  public run(error: UserError, { message }: CommandDeniedPayload) {
    if (Reflect.get(Object(error.context), 'silent')) return;
    return message.channel.send(error.message);
  }
}
```

:::note

In the code block above we use `if (Reflect.get(Object(error.context), 'silent'))` as opposed to `if (error.context.silent)` for TypeScript. When writing JavaScript code you can use the latter just fine.

To clarify this, with TypeScript `error.context` has the type `unknown` therefore trying to do `error.context.silent` will make TypeScript error on trying to read property `silent` of type `unknown`.

:::